cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -638,6 +646,8 @@ public void deleteArtifact(HttpRequest request, HttpResponder responder,
 
     try {
       artifactRepository.deleteArtifact(artifactId);
+      MetadataScope metadataScope = namespace.equals(Id.Namespace.SYSTEM) ? MetadataScope.SYSTEM : MetadataScope.USER;
```

Comment:
the handler should have as little logic as possible.  This should be moved into the ArtifactRepository.  If somebody else deletes using the repository, metadata should also be deleted.